tikv/dbreader/db_reader.go

```diff
@@ -38,6 +42,72 @@ type DBReader struct {
 	safePoint uint64
 }
 
+type RecordWithMeta struct {
+	StartTs  uint64
+	CommitTs uint64
+	Value    []byte
+}
+
+func (r *DBReader) GetWithMeta(key []byte, startTs uint64) (*RecordWithMeta, error) {
+	item, err := r.txn.Get(key)
+	if err != nil && err != badger.ErrKeyNotFound {
+		return nil, errors.Trace(err)
+	}
+	if err == badger.ErrKeyNotFound {
+		return nil, nil
+	}
+	dbUsrMeta := mvcc.DBUserMeta(item.UserMeta())
+	if dbUsrMeta.CommitTS() <= startTs {
+		val, err := item.Value()
```

Comment:
@coocood  Is it safe to use `item.Value` directly instead of `item.ValueCopy`

Reply:
No. The returned slice may be memory-mapped, if compaction deletes the backend table, related resources will be released after transaction discarded. Which means you will get a segment fault when access value.

Reply:
So we better change the `item.Value` usages in this file into `item.ValueCopy` in this file to make it safe ?  @coocood @bobotu

Reply:
Yes